### PR TITLE
feat: 282, 368, 284 Migrated MoD Handover backend functionality 

### PR DESCRIPTION
**Pull request by riginoommen**
# Resolves 282, 368, 284

### 

# Explain the feature
This PR provides the MoD Handover backend configuration which enables to talk to postgres database and salesforce 

- SFDC Integration.

- API Integration with the Database and graphql types.

## Does this PR introduce a breaking change

Yes

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Does the change have appropriate unit tests?
- [x] Did tests pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?

### CODE REVIEW

File: packages/mod-handover-service/src/typedef.graphql [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+type Query {
+  # Fetch all MoD HandOvers
+  listAllHandOvers: [HandOverType]
+  # Fetch Specific Handover by ID
+  listHandOver(handover_id: Float): HandOverType
+  # Fetch Handovers By Date
+  listHandOverByDate(date: String): [HandOverType]
+  # Fetch Cases Associated with the handover
+  listCasesByHandOver(handover_id: Float): [CaseType]
```
Comment (sayak-sarkar) [inline packages/mod-handover-service/src/typedef.graphql:151]:
```suggestion
  listCasesByHandover(handover_id: Float): [CaseType]
```

File: packages/mod-handover-service/src/typedef.graphql [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+  # Fetch Unassigned Count
+  listUnassignedCount: CountType
+  # Fetch the Platform Count
+  listAllPlatformCount: StatsCountType
+  # Fetch the ET Count
+  listAllETCount: StatsCountType
+  # Fetch the APS Count
+  listAllAPSCount: StatsCountType
+  # Fetch the MPS Count
+  listAllMPSCount: StatsCountType
```
Comment (sayak-sarkar) [inline packages/mod-handover-service/src/typedef.graphql:161]: @riginoommen Do we need these anymore?

File: packages/mod-handover-service/src/typedef.graphql [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+  # Fetch SBRs associated with the case no
+  listCaseSBRs(casenumber: String): [CaseShiftTrackerType]
+  # Fetch the total Handover Count
+  listTotalHandOverCount: CountType
```
Comment (sayak-sarkar) [inline packages/mod-handover-service/src/typedef.graphql:165]:
```suggestion
  listTotalHandoverCount: CountType
```

File: packages/mod-handover-service/src/typedef.graphql [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+  listSFDCCase(casenumber: String): SFDCType
+}
+
+type Mutation {
+  # Create a Handover in MoD Database
+  createHandOver(input: HandOverInput): HandOverType
+  # update a Handover in MoD Database
+  updateHandOver(input: HandOverInput): HandOverType
+  # Remove a Handover in MoD Database
+  removeHandOver(handover_id: Float!): HandOverType
```
Comment (sayak-sarkar) [inline packages/mod-handover-service/src/typedef.graphql:176]:
```suggestion
  createHandover(input: HandoverInput): HandoverType
  # update a Handover in MoD Database
  updateHandover(input: HandoverInput): HandoverType
  # Remove a Handover in MoD Database
  removeHandover(handover_id: Float!): HandoverType
```

File: packages/mod-handover-service/src/typedef.graphql [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+  # Create a Case in MoD Database
+  createCase(input: CaseInput): CaseType
+  # update a Case in MoD Database
+  updateCase(input: CaseInput): CaseType
+  # Remove a Case in MoD Database
+  removeCase(case_id: Int!): CaseType
+  # Send Handover Email
+  sendHandOverEmail(input: EmailInput): EmailType
```
Comment (sayak-sarkar) [inline packages/mod-handover-service/src/typedef.graphql:184]:
```suggestion
  sendHandoverEmail(input: EmailInput): EmailType
```

File: packages/mod-handover-service/src/types.d.ts [hunk 6]
```diff
@@ -0,0 +1,7 @@
+declare module '*.graphql';
+declare module '*.json';
+
+// define your types here
+type ModHandoverType = {
```
Comment (sayak-sarkar) [inline packages/mod-handover-service/src/types.d.ts:5]:
```suggestion
type MoDHandoverType = {
```

File: packages/mod-handover-service/README.md [hunk 7]
```diff
@@ -0,0 +1,25 @@
+ModHandover Microservice
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/README.md]: Changed

File: packages/mod-handover-service/README.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+=================================================
+
+One platform's server-side ModHandover GraphQL microservice.
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/README.md]: Changed

File: packages/mod-handover-service/Dockerfile [hunk 9]
```diff
@@ -0,0 +1,15 @@
+FROM node:alpine
+LABEL Name=one-platform-mod-handover-service \
+  Version=0.0.1 \
+  maintainer="mdeshmuk@redhat.com"
+
+# Building the user-group microservice
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/Dockerfile]: Changed

File: packages/mod-handover-service/.env.example [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+# DB_PASSWORD=
+# MOD_DB_NAME=
+# SFDC_DB_NAME=
+
+## SFDC Config
+# SFDC_USER=
+# SFDC_PASS=
+
+## Cookie Parser
+# COOKIE_PARSER=MOD-HANDOVER_SECRET
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/.env.example]: Changed as suggested

File: packages/mod-handover-service/src/typedef.graphql (hunk 2 above)
Comment (riginoommen) [reply packages/mod-handover-service/src/typedef.graphql]: Yes i removed it. But i may re add this becuse there is a discussion going on regarding it

File: packages/mod-handover-service/src/typedef.graphql (hunk 3 above)
Comment (riginoommen) [reply packages/mod-handover-service/src/typedef.graphql]: Changed as suggested

File: packages/mod-handover-service/package.json [hunk 11]
```diff
@@ -0,0 +1,95 @@
+{
+  "name": "mod-handover",
+  "description": "ModHandover micro-service for One Platform",
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/package.json]: Changed as suggested

File: packages/mod-handover-service/src/typedef.graphql (hunk 4 above)
Comment (riginoommen) [reply packages/mod-handover-service/src/typedef.graphql]: Changed as suggested

File: packages/mod-handover-service/src/typedef.graphql (hunk 5 above)
Comment (riginoommen) [reply packages/mod-handover-service/src/typedef.graphql]: Changed as suggested

File: packages/mod-handover-service/src/types.d.ts (hunk 6 above)
Comment (riginoommen) [reply packages/mod-handover-service/src/types.d.ts]: Changed as suggested

File: packages/mod-handover-service/package.json [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+  "opservice": {
+    "className": "ModHandover",
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/package.json]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 13]
```diff
@@ -0,0 +1,305 @@
+import dotenv from 'dotenv';
+/* Mock */
+import mock from './mock.json';
+import ModHandover from '../../service';
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 14]
```diff
@@ -0,0 +1,305 @@
...
+    unassigned_count
+    worh_count
+    irt_count
+    woc_count
+    unassigned_ncq_count
+    time
+  }
+
+  query ListAllHandOvers {
+    listAllHandOvers {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 15]
```diff
@@ -0,0 +1,305 @@
...
+    rme_count
+    unassigned_count
+    worh_count
+    irt_count
+    woc_count
+    unassigned_ncq_count
+    time
+  }
+
+  query ListAllHandOvers {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 16]
```diff
@@ -0,0 +1,305 @@
...
+    }
+  }
+
+  query ListAllMPSCount {
+    listAllMPSCount {
+      ...statsCountType
+    }
+  }
+
+  mutation CreateHandOver($input: HandOverInput) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 17]
```diff
@@ -0,0 +1,305 @@
...
+    createHandOver(input: $input) {
+      ...handOverType
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 18]
```diff
@@ -0,0 +1,305 @@
...
+  }
+
+  query ListAllMPSCount {
+    listAllMPSCount {
+      ...statsCountType
+    }
+  }
+
+  mutation CreateHandOver($input: HandOverInput) {
+    createHandOver(input: $input) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 19]
```diff
@@ -0,0 +1,305 @@
...
+      ...handOverType
+    }
+  }
+
+  mutation UpdateHandOver($input: HandOverInput) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 20]
```diff
@@ -0,0 +1,305 @@
...
+    updateHandOver(input: $input) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 21]
```diff
@@ -0,0 +1,305 @@
...
+      ...handOverType
+    }
+  }
+
+  mutation RemoveHandOver($handover_id: Float!) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 22]
```diff
@@ -0,0 +1,305 @@
...
+    removeHandOver(handover_id: $handover_id) {
+      ...handOverType
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 23]
```diff
@@ -0,0 +1,305 @@
...
+  }
+
+  mutation UpdateHandOver($input: HandOverInput) {
+    updateHandOver(input: $input) {
+      ...handOverType
+    }
+  }
+
+  mutation RemoveHandOver($handover_id: Float!) {
+    removeHandOver(handover_id: $handover_id) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 24]
```diff
@@ -0,0 +1,305 @@
...
+    worh_count
+    irt_count
+    woc_count
+    unassigned_ncq_count
+    time
+  }
+
+  query ListAllHandOvers {
+    listAllHandOvers {
+      ...handOverType
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/service.ts [hunk 25]
```diff
@@ -0,0 +1,50 @@
+import express from 'express';
+import { ApolloServer } from 'apollo-server-express';
+import http from 'http';
+const { ApolloLogExtension } = require( 'apollo-log' );
+import gqlSchema from './src/typedef.graphql';
+import { ModHandoverResolver as resolver } from './src/resolver';
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/service.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 26]
```diff
@@ -0,0 +1,305 @@
...
+
+  mutation CreateHandOver($input: HandOverInput) {
+    createHandOver(input: $input) {
+      ...handOverType
+    }
+  }
+
+  mutation UpdateHandOver($input: HandOverInput) {
+    updateHandOver(input: $input) {
+      ...handOverType
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 27]
```diff
@@ -0,0 +1,305 @@
...
+
+  mutation RemoveHandOver($handover_id: Float!) {
+    removeHandOver(handover_id: $handover_id) {
+      ...handOverType
+    }
+  }
+  `;
+
+beforeAll( () => {
+  request = supertest.agent( ModHandover );
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 28]
```diff
@@ -0,0 +1,305 @@
...
+} );
+afterAll( done => {
+  return ModHandover.close( done );
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 29]
```diff
@@ -0,0 +1,305 @@
...
+} );
+
+describe( 'ModHandover Microservice API Test', () => {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 30]
```diff
@@ -0,0 +1,305 @@
...
+  it( 'Returns All HandOvers List', ( done ) => {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 31]
```diff
@@ -0,0 +1,305 @@
...
+    request
+      .post( '/graphql' )
+      .send( {
+        query: query,
+        operationName: 'ListAllHandOvers'
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 32]
```diff
@@ -0,0 +1,305 @@
...
+      } )
+      .expect( ( res: any ) => {
+        expect( res.body ).not.toHaveProperty( 'errors' );
+        expect( res.body ).toHaveProperty( 'data' );
+
+        expect( res.body.data ).toHaveProperty( 'listAllHandOvers' );
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 33]
```diff
@@ -0,0 +1,305 @@
...
+        expect( res.body.data.listAllHandOvers[ 0 ] ).toHaveProperty( 'unassigned_ncq_sev_1_2_count' );
+        expect( res.body.data.listAllHandOvers[ 0 ] ).toHaveProperty( 'urgent_severity_1_breaches_count' );
+        expect( res.body.data.listAllHandOvers[ 0 ] ).toHaveProperty( 'manager_id' );
+        expect( res.body.data.listAllHandOvers[ 0 ] ).toHaveProperty( 'manager_first_name' );
+        expect( res.body.data.listAllHandOvers[ 0 ] ).toHaveProperty( 'manager_last_name' );
+        expect( res.body.data.listAllHandOvers[ 0 ] ).toHaveProperty( 'manager_email' );
+        expect( res.body.data.listAllHandOvers[ 0 ] ).toHaveProperty( 'handover_type' );
+        expect( res.body.data.listAllHandOvers[ 0 ] ).toHaveProperty( 'watchlist' );
+        expect( res.body.data.listAllHandOvers[ 0 ] ).toHaveProperty( 'proactive_cases' );
+        expect( res.body.data.listAllHandOvers[ 0 ] ).toHaveProperty( 'sensitive_accounts' );
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 34]
```diff
@@ -0,0 +1,305 @@
...
+        operationName: 'UpdateHandOver',
+        variables: {
+          input: mock
+        }
+      } )
+      .expect( ( res: any ) => {
+        expect( res.body ).not.toHaveProperty( 'errors' );
+        expect( res.body ).toHaveProperty( 'data' );
+
+        expect( res.body.data ).toHaveProperty( 'updateHandOver' );
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 35]
```diff
@@ -0,0 +1,305 @@
...
+        done( err );
+      } );
+  } );
+
+  it( 'Updates HandOver', ( done ) => {
+    request
+      .post( '/graphql' )
+      .send( {
+        query: query,
+        operationName: 'UpdateHandOver',
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 36]
```diff
@@ -0,0 +1,305 @@
...
+        operationName: 'CreateHandOver',
+        variables: {
+          input: mock
+        }
+      } )
+      .expect( ( res: any ) => {
+        expect( res.body ).not.toHaveProperty( 'errors' );
+        expect( res.body ).toHaveProperty( 'data' );
+
+        expect( res.body.data ).toHaveProperty( 'createHandOver' );
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 37]
```diff
@@ -0,0 +1,305 @@
...
+        done( err );
+      } );
+  } );
+
+  it( 'Creates New HandOver', ( done ) => {
+    request
+      .post( '/graphql' )
+      .send( {
+        query: query,
+        operationName: 'CreateHandOver',
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 38]
```diff
@@ -0,0 +1,305 @@
...
+        expect( res.body.data.listAllMPSCount ).toHaveProperty( 'woc_count' );
+        expect( res.body.data.listAllMPSCount ).toHaveProperty( 'unassigned_ncq_count' );
+        expect( res.body.data.listAllMPSCount ).toHaveProperty( 'time' );
+      } )
+      .end( ( err, res ) => {
+        done( err );
+      } );
+  } );
+
+  it( 'Creates New HandOver', ( done ) => {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 39]
```diff
@@ -0,0 +1,305 @@
...
+        expect( res.body ).toHaveProperty( 'data' );
+
+        expect( res.body.data ).toHaveProperty( 'updateHandOver' );
+      } )
+      .end( ( err, res ) => {
+        done( err );
+      } );
+  } );
+
+  it( 'Remove HandOver', ( done ) => {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 40]
```diff
@@ -0,0 +1,305 @@
...
+        query: query,
+        operationName: 'RemoveHandOver',
+        variables: {
+          handover_id: mock.handover_id
+        }
+      } )
+      .expect( ( res: any ) => {
+        expect( res.body ).not.toHaveProperty( 'errors' );
+        expect( res.body ).toHaveProperty( 'data' );
+        expect( res.body.data ).toHaveProperty( 'removeHandOver' );
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 41]
```diff
@@ -0,0 +1,736 @@
...
+import * as pg from 'pg';
+import moment from 'moment';
+
+const nodemailer = require( 'nodemailer' );
+
+const modConnectionString = `postgres://${ process.env.DB_USER }:${ process.env.DB_PASSWORD }@${ process.env.DB_HOST }/${ process.env.MOD_DB_NAME }`;
+
+const modSfdcConnectionString = `postgres://${ process.env.DB_USER }:${ process.env.DB_PASSWORD }@${ process.env.DB_HOST }/${ process.env.SFDC_DB_NAME }`;
+
+export const ModHandoverResolver = {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/e2e/mod-handover.spec.ts [hunk 42]
```diff
@@ -0,0 +1,305 @@
...
+        done( err );
+      } );
+  } );
+
+  it( 'Remove HandOver', ( done ) => {
+    request
+      .post( '/graphql' )
+      .send( {
+        query: query,
+        operationName: 'RemoveHandOver',
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/e2e/mod-handover.spec.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 43]
```diff
@@ -0,0 +1,736 @@
...
+  Query: {
+    listAllHandOvers ( root: any, args: any, ctx: any ) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 44]
```diff
@@ -0,0 +1,736 @@
...
+        .then( res => {
+          Client.end();
+          return res.rows;
+        } )
+        .catch( err => {
+          console.log( err );
+          return err;
+        } );
+    },
+    listHandOver ( root: any, args: any, ctx: any ) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 45]
```diff
@@ -0,0 +1,736 @@
...
+        .then( res => {
+          Client.end();
+          return res.rows;
+        } )
+        .catch( err => {
+          console.log( err );
+          return err;
+        } );
+    },
+    listCasesByHandOver ( root: any, args: any, ctx: any ) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 46]
```diff
@@ -0,0 +1,736 @@
...
+          } else {
+            return null;
+          }
+        } )
+        .catch( err => {
+          console.log( err );
+          return err;
+        } );
+    },
+    listTotalHandOverCount ( root: any, args: any, ctx: any ) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 47]
```diff
@@ -0,0 +1,736 @@
...
+        .then( res => {
+          Client.end();
+          return res.rows[ 0 ];
+        } )
+        .catch( err => {
+          console.log( err );
+          return err;
+        } );
+    },
+    listHandOverByDate ( root: any, args: any, ctx: any ) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 48]
```diff
@@ -0,0 +1,736 @@
...
+          console.log( err );
+          return err;
+        } );
+    },
+    listSFDCCase ( root: any, args: any, ctx: any ) {
+      return SFDCAPIHelper.getSFDCCase( args.casenumber );
+    },
+  },
+  Mutation: {
+    createHandOver ( root: any, args: any, ctx: any ) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 49]
```diff
@@ -0,0 +1,736 @@
...
+        .then( res => {
+          Client.end();
+          return args.input;
+        } )
+        .catch( err => {
+          console.log( err );
+          return err;
+        } );
+    },
+    updateHandOver ( root: any, args: any, ctx: any ) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 50]
```diff
@@ -0,0 +1,736 @@
...
+        .then( res => {
+          Client.end();
+          return args.input;
+        } )
+        .catch( err => {
+          console.log( err );
+          return err;
+        } );
+    },
+    removeHandOver ( root: any, args: any, ctx: any ) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 51]
```diff
@@ -0,0 +1,736 @@
...
+        .then( res => {
+          Client.end();
+          return args.input;
+        } )
+        .catch( err => {
+          console.log( err );
+          return err;
+        } );
+    },
+    sendHandOverEmail ( root: any, args: any, ctx: any ) {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 52]
```diff
@@ -0,0 +1,736 @@
...
+      let subject: String;
+      let content: String;
+      if ( args.input.handover.handover_type === 'Application Platform Support' ) {
+        subject = 'APS HandOver: ' + args.input.handover.handover_region + ' ' + new Date( args.input.handover.date ).toDateString();
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/resolver.ts [hunk 53]
```diff
@@ -0,0 +1,736 @@
...
+      } else if ( args.input.handover.handover_type === 'Enterprise Cloud Support' ) {
+        subject = 'ECS HandOver: ' + args.input.handover.handover_region + ' ' + new Date( args.input.handover.date ).toDateString();
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Yes both APS and ECS still actively uses the handover tool

File: packages/mod-handover-service/src/resolver.ts [hunk 54]
```diff
@@ -0,0 +1,736 @@
...
+      } else {
+        subject = 'MoD HandOver: ' + args.input.handover.handover_region + ' ' + new Date( args.input.handover.date ).toDateString();
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/resolver.ts]: Changed as suggested

File: packages/mod-handover-service/src/typedef.graphql [hunk 55]
```diff
@@ -0,0 +1,185 @@
+scalar DateTime
+
+type HandOverType {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/typedef.graphql]: Changed as suggested

File: packages/mod-handover-service/src/typedef.graphql [hunk 56]
```diff
@@ -0,0 +1,185 @@
...
+  handover: HandOverType
+  cases: [CaseType]
+  emailRecipient: [String]
+}
+
+type SFDCType {
+  account_name: String
+}
+
+input HandOverInput {
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/typedef.graphql]: Changed as suggested

File: packages/mod-handover-service/src/typedef.graphql [hunk 57]
```diff
@@ -0,0 +1,185 @@
...
+  emt_case_notes: String
+}
+
+input RecepientInput {
+  to: String
+  cc: String
+}
+
+input EmailInput {
+  handover: HandOverInput
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/typedef.graphql]: Changed as suggested

File: packages/mod-handover-service/src/typedef.graphql [hunk 58]
```diff
@@ -0,0 +1,185 @@
...
+  cases: [CaseInput]
+  emailRecipient: RecepientInput
+}
+
+type Query {
+  # Fetch all MoD HandOvers
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/typedef.graphql]: Changed as suggested

File: packages/mod-handover-service/src/typedef.graphql [hunk 59]
```diff
@@ -0,0 +1,185 @@
...
+  listAllHandOvers: [HandOverType]
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/typedef.graphql]: Changed as suggested

File: packages/mod-handover-service/src/typedef.graphql [hunk 60]
```diff
@@ -0,0 +1,185 @@
...
+  # Fetch Specific Handover by ID
+  listHandOver(handover_id: Float): HandOverType
```
[the remark this answers is not in the document]
Comment (riginoommen) [reply packages/mod-handover-service/src/typedef.graphql]: Changed as suggested

File: packages/mod-handover-service/src/typedef.graphql (hunk 1 above)
Comment (riginoommen) [reply packages/mod-handover-service/src/typedef.graphql]: Changed as suggested

**Pull request closed without merging**